packages/features/insights/server/events.ts

```diff
@@ -207,7 +207,39 @@ class EventsInsights {
       where: whereConditional,
     });
 
-    return csvData;
+    const csvDataWithBookerInfo = await Promise.all(
+      csvData.map(async (bookingTimeStatus) => {
+        if (!bookingTimeStatus.uid) {
+          return bookingTimeStatus;
+        }
+
+        const booking = await prisma.booking.findUnique({
```

Comment:
Instead of doing one query per row, can we instead findMany and merge afterwards? I see this becoming real slow on big data tables

Reply:
Great point, I opened a PR here! https://github.com/calcom/cal.com/pull/17281